modules/developer-cli-odo-debugging-an-application.adoc

```diff
@@ -10,11 +10,83 @@ You can debug your application on in `odo` with the `odo debug` command.
 
 .Procedure
 
-. After an application is deployed, start the port forwarding for your component to debug the application:
+. Download the sample application which contains the necessary `debugrun` step within its devfile:
++
+[source,terminal]
+----
+$ odo create nodejs --starter
+----
++
+[source,terminal]
+----
+Validation
+ ✓  Checking devfile existence [11498ns]
+ ✓  Checking devfile compatibility [15714ns]
+ ✓  Creating a devfile component from registry: DefaultDevfileRegistry [17565ns]
+ ✓  Validating devfile component [113876ns]
+
+Starter Project
+ ✓  Downloading starter project nodejs-starter from https://github.com/odo-devfiles/nodejs-ex.git [428ms]
+
+Please use `odo push` command to create the component with source deployed
+----
+
+. Push the application with the `--debug` flag which is required for all debugging deployments:
++
+[source,terminal]
+----
+$ odo push --debug
+----
++
+[source,terminal]
+----
+Validation
+ ✓  Validating the devfile [29916ns]
+
+Creating Kubernetes resources for component nodejs
+ ✓  Waiting for component to start [38ms]
+
+Applying URL changes
+ ✓  URLs are synced with the cluster, no changes are required.
+
+Syncing to component nodejs
+ ✓  Checking file changes for pushing [1ms]
+ ✓  Syncing files to the component [778ms]
+
+Executing devfile commands for component nodejs
+ ✓  Executing install command "npm install" [2s]
+ ✓  Executing debug command "npm run debug" [1s]
+
+Pushing devfile component nodejs
+ ✓  Changes successfully pushed to component
+----
++
+NOTE: You can specify a custom debug command by using the `--debug-command="custom-step"` flag.
```

Comment:
In the preview build I generated, the `NOTE` admonitions look fine, but our [guidelines](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#admonitions) are to format as such:

```
[IMPORTANT]
====
lorem ipsum
====
```
So it might be worth an update to these instances.